modules/tracktion_engine/playback/graph/tracktion_LoopingMidiNode.cpp

```diff
@@ -711,9 +711,13 @@ struct EventGenerator   : public MidiGenerator
                                 bool useMPEChannelMode, MPESourceID midiSourceID,
                                 juce::Array<juce::MidiMessage>& controllerMessagesScratchBuffer) override
     {
-        thread_local MidiMessageArray scratchBuffer, cleanedBufferToMerge;
+        ActiveNoteList originalState;
```

Comment:
Ok, I think we're getting close now but I can't see why this `originalState` is required?
It looks like you're just making a copy of `activeNoteList`, using `originalState` but then updating `activeNoteList`?

Is it something to avoid modifying the `activeNoteList` state whilst iterating and checking it? (As the non-modified `originalState` is now being used?) If so, what was the problem this was solving?